sqlalchemy_ibmi/base.py

```diff
@@ -1135,6 +1163,37 @@ def get_columns(self, connection, table_name, schema=None, **kw):
             })
         return sa_columns
 
+    def get_pk_constraint(self, connection, table_name, schema=None, **kw):
+        current_schema = self.denormalize_name(
+            schema or self.default_schema_name)
+        table_name = self.denormalize_name(table_name)
+        sysconst = self.sys_table_constraints
+        syskeyconst = self.sys_key_constraints
+
+        query = select([syskeyconst.c.colname, sysconst.c.tabname]).where(
+            and_(
+                syskeyconst.c.conschema == sysconst.c.conschema,
+                syskeyconst.c.conname == sysconst.c.conname,
+                sysconst.c.tabschema == current_schema,
+                sysconst.c.tabname == table_name,
+                sysconst.c.contype == 'PRIMARY KEY'
+            )
+        ).order_by(syskeyconst.c.colno)
+
+        result = connection.execute(query)
+        rows = result.fetchall()
+
+        if rows:
+            pk_columns = [self.normalize_name(row[0]) for row in rows]
+            pk_name = self.normalize_name(table_name + '_PK')  # Generate a default PK constraint name if needed
+
+            return {
+                "constrained_columns": pk_columns,
+                "name": pk_name
+            }
+
+        return None
```

Comment:
```suggestion
        return {"constrained_columns": [], "name": None}
```

Looking at the other dialects, this must always return a value.